graphql_compiler/compiler/emit_sql.py

```diff
@@ -182,6 +182,21 @@ def _find_folded_fields(ir: IrAndMetadata) -> Dict[FoldPath, Set[FoldScopeLocati
     return folded_fields
 
 
+class SQLFoldTraversalDescriptor(NamedTuple):
```

Comment:
Any particular reason to use a NamedTuple and not a dataclass here?

Reply:
Oh this seems to just be moved from elsewhere in this file, never mind. The case for a dataclass is still good, but since this is existing code, it doesn't need to happen here.